pkg/backup/item_backupper.go

```diff
@@ -178,30 +182,36 @@ func (ib *defaultItemBackupper) backupItem(logger logrus.FieldLogger, obj runtim
 		return err
 	}
 
-	backupErrs := make([]error, 0)
-	err = ib.executeActions(log, obj, groupResource, name, namespace, metadata)
-	if err != nil {
+	var (
+		backupErrs       []error
+		pod              *corev1api.Pod
+		podVolumeBackups []string
+	)
+
+	if groupResource == kuberesource.Pods {
+		// get the volumes to backup using restic, and add any of them that are PVCs to the pvc snapshot
+		// tracker, so that when we backup PVCs/PVs via an item action in the next step, we don't snapshot
+		// PVs that will have their data backed up with restic.
+		if pod, podVolumeBackups, err = getPodVolumesToBackup(obj, ib.pvcSnapshotTracker); err != nil {
```

Comment:
Having `getPodVolumesToBackup` also add to the tracker seems like it's doing more work than it should. What if we did this?
```go
pod, err = podFromUnstructured(obj)
// check err

resticVolumesToBackup := restic.GetVolumesToBackup(pod)
ib.pvcSnapshotTracker.Track(pod, resticVolumesToBackup) // this has the looping logic
```

(I think `resticVolumesToBackup` is more descriptive than `podVolumeBackups`.)

Reply:
Good idea. I think I was tired when I wrote this. updated.